### PR TITLE
feat: make application usage private through project

### PR DESCRIPTION
**Pull request by sjaanus**

### CODE REVIEW

File: src/lib/services/client-metrics/instance-service.ts [hunk 1]
```diff
@@ -162,8 +174,27 @@ export default class ClientInstanceService {
 
     async getApplications(
         query: IApplicationQuery,
+        userId: number,
     ): Promise<IClientApplication[]> {
-        return this.clientApplicationsStore.getAppsForStrategy(query);
+        const applications =
+            await this.clientApplicationsStore.getAppsForStrategy(query);
+        if (this.flagResolver.isEnabled('privateProjects') && userId) {
```
Comment (FredrikOseberg) [inline src/lib/services/client-metrics/instance-service.ts:181]: Can userId be 0?

**Review by FredrikOseberg — CHANGES_REQUESTED** [review]
Looks good, but are we missing a test for this?

File: src/lib/services/client-metrics/instance-service.ts (hunk 1 above)
Comment (sjaanus) [reply src/lib/services/client-metrics/instance-service.ts:181]: That is good question. So far I have discovered we can have -1 userId, but have not seen 0.

**Review by FredrikOseberg — APPROVED** [review]
LGTM

**Pull request merged**